x509/codesigning_cert_validations.go

```diff
@@ -34,12 +34,12 @@ func ValidateCodeSigningCertChain(certChain []*x509.Certificate, signingTime *ti
 	// For self-signed signing certificate (not a CA)
 	if len(certChain) == 1 {
 		cert := certChain[0]
+		if err := validateSelfSignedCert(cert, true); err != nil {
+			return err
+		}
```

Comment:
Can we just add an issuer check here? Something like
```
if !bytes.Equal(cert.RawSubject, cert.RawIssuer) {
    return fmt.Errorf("self-signed certificate with subject %q is not self-issued. The issuer is %q", cert.Subject, cert.Issuer)
}
```
If the logic is used more than one places, we could have a helper function named `isSelfIssued` in `helper.go`.

Reply:
Updated.